### PR TITLE
add scale test for node out-of-service-detach feature

### PR DESCRIPTION
**Pull request by sonasingh46**
Add scale test job for node out of service detach feature. 
Ref test PR: https://github.com/kubernetes/kubernetes/pull/118848

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml [hunk 1]
```diff
@@ -1007,3 +1007,52 @@ periodics:
   annotations:
     testgrid-dashboards: google-gce
     testgrid-tab-name: soak-gci-gce-1.12
+
+  - name: pull-kubernetes-e2e-gce-nodeoutofservicedetach
```
Comment (sonasingh46) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:1011]: @xing-yang -- Not sure, what should the approach to run this test in this case. This job only runs the node out of service feature scale test. Is this the right approach?

**Comment by xing-yang** [comment]
The new test `pull-kubernetes-e2e-gce-nodeoutofservicedetach` should be added to the testgrid similar to this one:
 https://github.com/kubernetes/test-infra/blob/master/config/testgrids/kubernetes/presubmits/config.yaml#L96

**Comment by xing-yang** [comment]
assign @gnufied @msau42 @mattcary

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml [hunk 2]
```diff
@@ -371,6 +371,54 @@ presubmits:
...
+        - --env=KUBE_PROXY_DAEMONSET=true
+        - --env=ENABLE_POD_PRIORITY=true
+        - --env=ENABLE_APISERVER_TRACING=true
+        - --extract=local
+        - --gcp-node-image=gci
+        - --gcp-zone=us-west1-b
+        - --provider=gce
+        - --runtime-config=api/all=true
+        - --stage=gs://kubernetes-release-pull/ci/pull-kubernetes-e2e-gce-nodeoutofservicedetach-scale-disruptive
+        - --gcp-nodes=4
```
Comment (sonasingh46) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:410]: I had earlier used `--num-nodes` here but then made it `--gcp-nodes` 
Reason: I think prow still uses kubetest and kubetest2 and figured that `--num-nodes` is only available in kubetest2.

**Comment by sonasingh46** [comment]
Once this merges and this test passes on the [PR](https://github.com/kubernetes/kubernetes/pull/118848/files), I can raise one PR to make it periodic and remove this from pre submit. 
Is this the right approach?

**Comment by xing-yang** [comment]
/hold
We are not going to merge the [PR](https://github.com/kubernetes/kubernetes/pull/118848) because it depends on a specific cloud provider.

**Pull request closed without merging**